itou/users/tests/test_models.py

```diff
@@ -800,17 +800,17 @@ def test_get_kind_display(self):
         assert "inspecteur du travail" == labor_inspector.get_kind_display()
 
     def test_constraint_user_lack_of_nir_reason_or_nir(self):
-        no_nir_user = JobSeekerFactory(nir=None)
+        no_nir_user = JobSeekerFactory(nir="")
         # This works
-        assert no_nir_user.nir is None
+        assert not no_nir_user.nir
         no_nir_user.lack_of_nir_reason = LackOfNIRReason.TEMPORARY_NUMBER
         no_nir_user.save()
 
         user = JobSeekerFactory()
         # This doesn't
         assert user.nir
         user.lack_of_nir_reason = LackOfNIRReason.TEMPORARY_NUMBER
-        with pytest.raises(IntegrityError):
+        with pytest.raises(ValidationError):
             user.save()
```

Comment:
On peut en profiter pour ajouter un `match`.
```suggestion
        with pytest.raises(
            ValidationError,
            match="Un utilisateur ayant un NIR ne peut avoir un motif justifiant l'absence de son NIR.",
        ):
            user.save()
```